static/app/views/explore/toolbar/styles.tsx

```diff
@@ -28,3 +31,47 @@ export const ToolbarHeading = styled('h6')<{disabled?: boolean}>`
 export const ToolbarHeaderButton = styled(Button)<{disabled?: boolean}>`
   color: ${p => (p.disabled ? p.theme.gray300 : p.theme.purple300)};
 `;
+
+export function ToolbarRow<T>({
+  index = 0,
+  rows = [],
+  setRows,
+  children,
+}: {
+  children: React.ReactNode;
+  index?: number;
+  rows?: T[];
+  setRows?: (rows: T[]) => void;
+}) {
+  const removeRow = useCallback(
+    (i: number) => {
+      const newRow = rows.filter((_, rowIndex) => rowIndex !== i);
+      setRows?.(newRow);
+    },
+    [setRows, rows]
+  );
+
+  return (
+    <ToolbarRowWrapper>
+      {children}
+      {rows.length > 1 ? (
```

Comment:
Instead of hiding, we should disable the button with `disable={rows.length <= 1}`

Reply:
Sure yup